yjit/src/codegen.rs

```diff
@@ -6088,7 +6097,9 @@ fn gen_send_cfunc(
         let expected_stack_after = asm.ctx.get_stack_size() as i32 - argc;
         if let Some(known_cfunc_codegen) = lookup_cfunc_codegen(unsafe { (*cme).def }) {
             // We don't push a frame for specialized cfunc codegen, so the generated code must be leaf.
-            if asm.with_leaf_ccall(|asm|
+            // However, the interpreter doesn't push a frame on opt_* instruction either, so we allow
+            // non-send instructions to break this rule as an exception.
+            if asm.with_leaf_ccall(jit.on_send_insn(), |asm|
```

Comment:
I feel like this could be refactored to either call or not call `asm.with_leaf_call` instead of passing a boolean parameter which makes the function not do what the name implies 😅

Reply:
I was thinking the latter. I think it's ok sometimes to repeat a bit of code (one expression here really). It just seems weird to have a function called `with_leaf_call` and then passing a boolean that says "well, actually not" 😅

Reply:
:+1: https://github.com/ruby/ruby/pull/10033/commits/5955a44f2373db12dbf85d6bf871a4238cea7734